core/trino-main/src/main/java/io/trino/cost/TableScanStatsRule.java

```diff
@@ -59,7 +60,7 @@ public TableScanStatsRule(Metadata metadata, StatsNormalizer normalizer)
     protected Optional<PlanNodeStatsEstimate> doCalculate(TableScanNode node, StatsProvider sourceStats, Lookup lookup, Session session, TypeProvider types)
     {
         // TODO Construct predicate like AddExchanges's LayoutConstraintEvaluator
```

Comment:
This is the reason why `SHOW STATS` for arbitrary query can have different results than it used to have when were calling `metadata. getTableStatistics()` directly in `ShowStatsQueryRewrite`.

IMO it is a pity that we don't pass a real constraint here. Please create an issue for that.

Reply:
discussed offline